src/main/java/emu/grasscutter/server/packet/recv/HandlerQuestUpdateQuestVarReq.java

```diff
@@ -20,28 +16,15 @@ public class HandlerQuestUpdateQuestVarReq extends PacketHandler {
     @Override
     public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
         //Client sends packets. One with the value, and one with the index and the new value to set/inc/dec
-        var req = QuestUpdateQuestVarReqOuterClass.QuestUpdateQuestVarReq.parseFrom(payload);
+        QuestUpdateQuestVarReq req = QuestUpdateQuestVarReq.parseFrom(payload);
         GameMainQuest mainQuest = session.getPlayer().getQuestManager().getMainQuestById(req.getQuestId()/100);
-        List<QuestVarOpOuterClass.QuestVarOp> questVars = req.getQuestVarOpListList();
-        if (mainQuest.getQuestVarsUpdate().size() == 0) {
-            for (QuestVarOpOuterClass.QuestVarOp questVar : questVars) {
-                mainQuest.getQuestVarsUpdate().add(questVar.getValue());
-            }
-        } else {
-            for (QuestVarOpOuterClass.QuestVarOp questVar : questVars) {
-                if (questVar.getIsAdd()) {
-                    if (questVar.getValue() >= 0) {
-                        mainQuest.incQuestVar(questVar.getIndex(), questVar.getValue());
-                    } else {
-                        mainQuest.decQuestVar(questVar.getIndex(), questVar.getValue());
-                    }
-                } else {
-                    mainQuest.setQuestVar(questVar.getIndex(), mainQuest.getQuestVarsUpdate().get(0));
-                }
-                //remove the first element from the update list
-                mainQuest.getQuestVarsUpdate().remove(0);
+
+        req.getQuestVarOpListList().forEach(questVar -> {
+            if (questVar.getIsAdd()) {
+                mainQuest.incQuestVar(questVar.getIndex(), questVar.getValue());
             }
-        }
+            mainQuest.decQuestVar(questVar.getIndex(), questVar.getValue());
```

Comment:
wouldn't it execute both here in case it adds, kinda negating the effect?

Are you sure its really not set whats needed to be used in the else?
The value is explicitly not unsigned in the proto, so instead of having add and decrement, they probably just add a negative value.
Also are you sure the stuff from the comment from the top and the questVarsUpdate logic can be safelly ignored?